lib/spack/docs/packaging_guide.rst

```diff
@@ -574,7 +574,9 @@ Download caching
 Spack maintains a cache (described :ref:`here <caching>`) which saves files
 retrieved during package installations to avoid re-downloading in the case that
 a package is installed with a different specification (but the same version) or
-reinstalled on account of a change in the hashing scheme.
+reinstalled on account of a change in the hashing scheme. It may
+(rarely) be necessary to avoid caching for a particular version by adding
```

Comment:
Could you elaborate on the cases where this would be useful?

* A tagged commit for a VCS is updated to point to a new commit (e.g. for a package in development)
* A package version refers to an archive URL that is updated (e.g. because the version is in development)